knotx-core/knotx-service-adapter-http/src/main/java/com/cognifide/knotx/core/serviceadapter/http/HttpClientFacade.java

```diff
@@ -55,33 +61,74 @@
-              request.method()));
+    final String servicePath = UriTransformer.resolveServicePath(params.getString(PATH_PROPERTY_KEY), originalRequest);
 
-      return serviceResponse.flatMap(item -> transformResponse(item, request.method(), request.path()));
+    final Optional<HttpServiceAdapterConfiguration.ServiceMetadata> serviceMetadata = findServiceMetadata(servicePath);
+    if (serviceMetadata.isPresent()) {
+      final HttpRequestWrapper serviceRequestWrapper = new HttpRequestWrapper(originalRequest.toJson());
+      serviceRequestWrapper.setPath(servicePath);
+      serviceRequest = Pair.of(serviceRequestWrapper, serviceMetadata.get());
     } else {
-      LOGGER.error("Could not handle request with path [{}]", request.path());
-      return Observable.just(new HttpResponseWrapper().setStatusCode(HttpResponseStatus.BAD_REQUEST));
+      serviceRequest = Pair.of(new HttpRequestWrapper().setPath(servicePath), null);
+    }
+
+    return serviceRequest;
+  }
+
+  private Optional<HttpServiceAdapterConfiguration.ServiceMetadata> findServiceMetadata(String servicePath) {
+    return services.stream().filter(metadata -> servicePath.matches(metadata.getPath())).findAny();
+  }
+
+  private void ensureRequestIsSupported(Pair<HttpRequestWrapper, HttpServiceAdapterConfiguration.ServiceMetadata> serviceRequest) {
+    final boolean anyServiceSupportsPath = serviceRequest.getRight() != null;
+    if (!anyServiceSupportsPath) {
+      final String error = String.format("Parameter `params.path`: `%s` not supported!", serviceRequest.getLeft().path());
+      throw new UnsupportedServiceException(error);
     }
   }
 
-  private Observable<HttpClientResponse> request(HttpClient client, HttpMethod method, int port, String domain, String uri, Action1<HttpClientRequest> requestBuilder) {
+  private Observable<HttpClientResponse> callService(Pair<HttpRequestWrapper, HttpServiceAdapterConfiguration.ServiceMetadata> serviceRequest) {
+    final HttpRequestWrapper requestWrapper = serviceRequest.getLeft();
+    final HttpServiceAdapterConfiguration.ServiceMetadata serviceMetadata = serviceRequest.getRight();
+
     return Observable.create(subscriber -> {
-      HttpClientRequest req = client.request(method, port, domain, uri);
-      Observable<HttpClientResponse> resp = req.toObservable();
+      HttpClientRequest request = httpClient.get(serviceMetadata.getPort(), serviceMetadata.getDomain(), requestWrapper.path());
+      Observable<HttpClientResponse> resp = request.toObservable();
       resp.subscribe(subscriber);
-      requestBuilder.call(req);
-      req.end();
+      request.headers().addAll(getFilteredHeaders(requestWrapper.headers(), serviceMetadata.getAllowedRequestHeaderPatterns()));
+
+      if (HttpMethod.POST == requestWrapper.method()) {
+        Buffer buffer = createFormPostBody(requestWrapper.formAttributes());
```

Comment:
We don't support POSTs in this iteration. I've removed it to avoid confusion.
As I understand, all parameters will be pased in path with GET method.